core/src/main/java/bio/terra/pearl/core/dao/study/StudyEnvironmentDao.java

```diff
@@ -106,4 +106,24 @@ public StudyEnvironment loadWithAllContent(StudyEnvironment studyEnv) {
     public void deleteByStudyId(UUID studyId) {
         deleteByProperty("study_id", studyId);
     }
+
+    /**
+     * Returns the StudyEnvironment matching the given Pepper study name and environment name.
+     */
+    public Optional<StudyEnvironment> findByPepperStudyName(String pepperStudyName, EnvironmentName environmentName) {
```

Comment:
Two thoughts:
1. This should be renamed to `findByStudyName(String studyName, ...)`. since that's what it's doing.  it's a service-level concern that pepperStudyName happens to coincide with study name
2. This should probably use study shortcode instead of name.  shortcode is guaranteed to be globally unique, whereas two studies can have the same name

Reply:
Good catch! Yes, there is a bug in the DAO _and_ in the test. My intent was to use `pepperStudyName`, so this DAO method name is correct, but the implementation is wrong. However, based on further discussion with Pegah, it looks like we're going to use `shortcode` (or a simple derivative of `shortcode`) instead, so no extra field needed, and this DAO method will go away because we already have `findByStudy`.